packages/contracts/contracts/Router.sol

```diff
@@ -0,0 +1,220 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.4;
+
+import "./interfaces/ITransactionManager.sol";
+import "./lib/LibAsset.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+contract Router is Ownable {
+  address public immutable routerFactory;
+
+  ITransactionManager public transactionManager;
+
+  uint256 private chainId;
```

Comment:
This has to match the transaction manager's `chainId`, not necessarily the canonical chain of this contract

Reply:
Shouldn't it match both

Reply:
depends on the underlying chain, some return `1` and it's not mainnet (i.e. Harmony)

Reply:
here we are getting the chainId from RouterFactory, In RouterFactory we are getting it from TransactionManager only.